src/mesh/voxel_cloud.rs

```diff
@@ -463,6 +481,322 @@ impl ScalarField {
         tools::weld(&joined_voxel_mesh, (min_voxel_dimension as f32) / 4.0)
     }
 
+    /// Computes boolean intersection (logical AND operation) of the current and
+    /// another scalar field. The current scalar field will be mutated and
+    /// resized to the size and position of an intersection of the two scalar
+    /// fields' volumes. The two scalar fields do not have to contain voxels of
+    /// the same size.
+    ///
+    /// The `volume_value_range` is an interval defining which values of the
+    /// scalar field should be considered to be a volume. The
+    /// `ScalarField::from_mesh` generates a scalar field, which marks volume
+    /// voxels with value `0`. `compute_distance_field` marks each voxel with a
+    /// value representing the voxel's distance from the original volume,
+    /// therefore the voxels right at the shell of the volume are marked 0, the
+    /// layer around them is marked 1 or -1 (inside closed volumes) etc. Once
+    /// the scalar field is populated with meaningful values, it is possible to
+    /// treat (perform boolean operations or materialize into mesh) on various
+    /// numerical ranges. Such range is specified ad-hoc by parameter
+    /// `volume_value_range`.
```

Comment:
you requested and explanation of the `volume_value_range`, so i've added them whereever it appears. it doesn't appear anywhere in the struct itself and it is basically a way of interpreting the values stored in it. are you sure it belongs to the description of the struct?
i don't mind, just asking

Reply:
It would be good if it were either module level or struct level documentation. Have a look at std or any other crate to see how to write module level documentation.